controllers/service_controller_test.go

```diff
@@ -0,0 +1,222 @@
+          }`,
+					},
+				},
+				{
+					JWTClaimSet: &v1beta1.Authorization_JWTClaimSet{
+						Match: &v1beta1.Authorization_JWTClaimSet_Match{
+							Http: &v1beta1.Authorization_JWTClaimSet_HTTPMatch{
+								Path: "/api/*",
+							},
+						},
+						Claim: &v1beta1.Authorization_JWTClaimSet_Claim{
+							Aud: "api",
+						},
+					}},
+			},
+		},
+		Status: v1beta1.ServiceStatus{
+			Ready: false,
+		},
+	}
+
+	secret = v1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Secret",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "secret",
+			Namespace: "authorino",
+		},
+		Data: map[string][]byte{
+			"clientID":     []byte("clientID"),
+			"clientSecret": []byte("clientSecret"),
+		},
+	}
+)
+
+func setupEnvironment(t *testing.T) ServiceReconciler {
+	scheme := runtime.NewScheme()
+	v1beta1.AddToScheme(scheme)
+	v1.AddToScheme(scheme)
+	// Create a fake client with a service and a secret.
+	client := fake.NewFakeClientWithScheme(scheme, &service, &secret)
+
+	c := cache.NewCache()
+
+	return ServiceReconciler{
+		Client: client,
+		Log:    ctrl.Log.WithName("reconcilerTest"),
+		Scheme: nil,
+		Cache:  &c,
+	}
+
+}
+
+func TestReconcilerOk(t *testing.T) {
+
+	r := setupEnvironment(t)
+
+	result, err := r.Reconcile(controllerruntime.Request{
+		NamespacedName: types.NamespacedName{
+			Namespace: service.Namespace,
+			Name:      service.Name,
+		},
+	})
+
+	if err != nil {
+		t.Error(err)
+	}
+
+	serviceCheck := v1beta1.Service{}
+
+	r.Client.Get(context.TODO(), client.ObjectKey{
+		Namespace: service.Namespace,
+		Name:      service.Name,
+	}, &serviceCheck)
+
+	assert.Check(t, serviceCheck.Status.Ready)
+	// Result should be empty
+	assert.DeepEqual(t, result, ctrl.Result{})
+}
+
+func TestReconcilerMissingSecret(t *testing.T) {
+
+	r := setupEnvironment(t)
+
+	r.Client.Delete(context.TODO(), &secret)
+
+	result, err := r.Reconcile(controllerruntime.Request{
+		NamespacedName: types.NamespacedName{
+			Namespace: service.Namespace,
+			Name:      service.Name,
+		},
+	})
+
+	// Error should be "secret" not found.
+	assert.Check(t, errors.IsNotFound(err))
+	// Result should be empty
+	assert.DeepEqual(t, result, ctrl.Result{})
+}
+
+func TestReconcilerNotFound(t *testing.T) {
+
+	r := setupEnvironment(t)
+
+	// Let's try to reconcile a non existing object.
+	result, err := r.Reconcile(controllerruntime.Request{
+		NamespacedName: types.NamespacedName{
+			Namespace: service.Namespace,
+			Name:      "nonExistant",
+		},
+	})
+
+	if err != nil {
+		t.Error(err)
+	}
+
+	serviceCheck := v1beta1.Service{}
+	r.Client.Get(context.TODO(), client.ObjectKey{
+		Namespace: service.Namespace,
+		Name:      service.Name,
+	}, &serviceCheck)
+
+	// The object we created should remain not ready
+	assert.Check(t, !serviceCheck.Status.Ready)
+
+	// Result should be empty
+	assert.DeepEqual(t, result, ctrl.Result{})
+}
+
+func TestTranslateService(t *testing.T) {
```

Comment:
This one left empty on purpose? Maybe a comment regarding the assertion to be made? or simply remove it